cmd/kube-controller-manager/app/autoscaling.go

```diff
@@ -48,9 +50,10 @@ func startHPAController(ctx ControllerContext) (http.Handler, bool, error) {
 
 func startHPAControllerWithRESTClient(ctx ControllerContext) (http.Handler, bool, error) {
 	clientConfig := ctx.ClientBuilder.ConfigOrDie("horizontal-pod-autoscaler")
+	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(clientConfig)
 	metricsClient := metrics.NewRESTMetricsClient(
 		resourceclient.NewForConfigOrDie(clientConfig),
-		custom_metrics.NewForConfigOrDie(clientConfig),
+		custom_metrics.NewForConfigOrDie(clientConfig, custom_metrics.APIVersionsFromDiscovery{DiscoveryClient: *discoveryClient}),
```

Comment:
this is going to be checking discovery for *every* metrics request.  We should be caching the information and periodically rechecking it, or rechecking if we don't have any results.